src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -53,39 +58,83 @@
+            List<Action> actions = new ArrayList<>();
+            if (step.getWait()) {
+                StepContext context = getContext();
+                actions.add(new BuildTriggerAction(context, step.isPropagate()));
+                LOGGER.log(Level.FINER, "scheduling a build of {0} from {1}", new Object[]{project, context});
+            }
+            actions.add(new CauseAction(new Cause.UpstreamCause(invokingRun)));
+            List<ParameterValue> parameters = step.getParameters();
+            if (parameters != null) {
+                parameters = completeDefaultParameters(parameters, (Job) project);
+                actions.add(new ParametersAction(parameters));
+            }
+            Integer quietPeriod = step.getQuietPeriod();
+            // TODO use new convenience method in 1.621
+            if (quietPeriod == null) {
+                quietPeriod = project.getQuietPeriod();
+            }
+            QueueTaskFuture<?> f = new ParameterizedJobMixIn() {
+                @Override
+                protected Job asJob() {
+                    return (Job) project;
+                }
+            }.scheduleBuild2(quietPeriod, actions.toArray(new Action[actions.size()]));
+            if (f == null) {
+                throw new AbortException("Failed to trigger build of " + project.getFullName());
             }
-        }.scheduleBuild2(quietPeriod, actions.toArray(new Action[actions.size()]));
-        if (f == null) {
-            throw new AbortException("Failed to trigger build of " + project.getFullName());
+        } else if (item instanceof Queue.Task){
+            if (step.getParameters() != null && !step.getParameters().isEmpty()) {
+                throw new AbortException("Item type does not support parameters");
+            }
+            Queue.Task task = (Queue.Task) item;
+            node.addAction(new LabelAction(Messages.BuildTriggerStepExecution_building_(task.getFullDisplayName())));
+            List<Action> actions = new ArrayList<>();
+            if (step.getWait()) {
+                StepContext context = getContext();
+                actions.add(new BuildTriggerAction(context, step.isPropagate()));
+                LOGGER.log(Level.FINER, "scheduling a build of {0} from {1}", new Object[]{task, context});
+            }
+            actions.add(new CauseAction(new Cause.UpstreamCause(invokingRun)));
+            Integer quietPeriod = step.getQuietPeriod();
+            if (quietPeriod == null) {
+                try {
+                    Method getQuietPeriod = task.getClass().getMethod("getQuietPeriod");
+                    if (getQuietPeriod.getReturnType().equals(int.class)) {
+                        quietPeriod = (Integer) getQuietPeriod.invoke(task);
+                    }
+                } catch (NoSuchMethodException | IllegalAccessError | IllegalArgumentException | InvocationTargetException e) {
```

Comment:
Only `NoSuchMethodException` should be ignored; other exceptions should be logged.